spk/sonarr/src/service-setup.sh

```diff
@@ -0,0 +1,88 @@
+
+# Sonarr service setup
+SONARR="${SYNOPKG_PKGDEST}/share/Sonarr/bin/Sonarr"
+
+# Sonarr uses custom Config and PID directories
+HOME_DIR="${SYNOPKG_PKGVAR}"
+CONFIG_DIR="${HOME_DIR}/.config"
+SONARR_CONFIG_DIR="${CONFIG_DIR}/Sonarr"
+PID_FILE="${SONARR_CONFIG_DIR}/sonarr.pid"
+CMD_ARGS="-nobrowser -data=${SONARR_CONFIG_DIR}"
+
+LEGACY_SPK_NAME="nzbdrone"
+LEGACY_SYNOPKG_PKGDEST="/var/packages/${LEGACY_SPK_NAME}/target"
+# check for legacy package data storage location
+if [ -d /var/packages/${LEGACY_SPK_NAME}/var ]; then
+    LEGACY_SYNOPKG_PKGVAR="/var/packages/${LEGACY_SPK_NAME}/var"
+else
+    LEGACY_SYNOPKG_PKGVAR="${LEGACY_SYNOPKG_PKGDEST}/var"
+fi
```

Comment:
You could use `SYNOPKG_DSM_VERSION_MAJOR` instead to detect if being on DSM7 or later, or not.  This would satisfy your check.  Also I'd recommend doing a test -d on both cases  else either exit 1 or considering it isn't installed.

I was looking at https://global.download.synology.com/download/Document/Software/DeveloperGuide/Os/DSM/All/enu/DSM_Developer_Guide_7_enu.pdf and wondered if you could check if a package is installed or not but couldnt see any of interest in scripting variables at page 56.

Further testing `synopkg status` does work, not entirely as expected but does work:
```
$ synopkg status ffmpeg
{"aspect":{"active":{"status":"stop","status_code":263,"status_description":"failed to get unit status"}},"description":"Status: [263], package is stopped","package":"ffmpeg","status":"stop"}
th0ma7@th0ma7-nas:~$ echo $?
7
th0ma7@th0ma7-nas:~$ synopkg status toto
{"aspect":{"active":{"status":"stop","status_code":263,"status_description":"failed to get unit status"},"error":{"status":"non_installed","status_code":255,"status_description":"failed to locate given package"}},"description":"Status: [255], No such package","package":"toto","status":"non_installed"}
th0ma7@th0ma7-nas:~$ echo $?
255
```

Reply:
I'm actually only interested in locating previous configuration files to migrate them. That is unless they are from Sonarr v2. I don't believe that checking the current DSM version will assist with that.

Regarding the recommendation to test with `-d` on the folders... I would normally agree with this however the tests later on for the specific files in `LEGACY_SYNOPKG_PKGDEST` and `LEGACY_CONFIG_DIR` are the substantive tests required. I wouldn't want to potentially have a null value for `LEGACY_SYNOPKG_PKGVAR` at this point and potentially invalidate those tests.

For the check for the presence of the older package, this would be ideal but given that it is a different package name the more direct method would be to check for the presence of the executable. This is done in the `validate_preinst` and is only done to exclude the upgrade from Sonarr v2.